Plugin/src/SofaPython3/PythonFactory.cpp

```diff
@@ -273,12 +292,34 @@ void PythonFactory::fromPython(BaseData* d, const py::object& o)
     if(!nfo.Container())
     {
         scoped_writeonly_access guard(d);
-        if(nfo.Integer())
+        if(nfo.Integer()) {
             nfo.setIntegerValue(guard.ptr, 0, py::cast<int>(o));
-        else if(nfo.Text())
+        } else if(nfo.Text()) {
             nfo.setTextValue(guard.ptr, 0, py::cast<py::str>(o));
-        else if(nfo.Scalar())
+        } else if(nfo.Scalar()) {
             nfo.setScalarValue(guard.ptr, 0, py::cast<double>(o));
+        } else {
+            // Custom datatype that has no DataTypeInfo. Let's see if we can convert it to a string
```

Comment:
Welll, this is open question but I'm not very fond of going through a string to convert the data. I prefer we spend more time implementing a proper data conversion through AbstracTypeInfo or any other general  mecanism.

Reply:
>  I'm not very fond of going through a string to convert the data

Watch out there, you are talking about the ground base of the ObjectFactory ;-) which sucks a lot because we can't use `node.addObject("MechanicalObject", position=myHugeNumpyArray.tolist())` without having to wait an eternity for it to convert it to a string and then to an array...

Anyway, I'm not sure what to do for this here, since some Data are not that trivial to convert to a compatible `AbstracTypeInfo`. I did this change for the BoundingBox, since it is an object that has two "array" (min and max), hence it is not a `nfo.Container`. Since all Data are supposed to be able to construct from a string, this change will allow us to write:
```python
root.bbox = [x0, y0, z0, x1, y1, z2]
```

I guess we could also do
```python
root.bbox.read(str([x0, y0, z0, x1, y1, z2]))
```
but it add yet another way to assign values to data, which start to be confusing...